.pipelines/pipeline.yaml

```diff
@@ -49,98 +49,12 @@ stages:
-                { { { {
-                      sudo -E env "PATH=$PATH" make test-all;
-                      echo $? >&3;
-                      } | tee >(build/tools/bin/go-junit-report > report.xml) >&4;
-                    } 3>&1;
-                  } | { read xs; exit $xs; }
-                } 4>&1
-              retryCountOnTaskFailure: 3
-              name: "Test"
-              displayName: "Run Tests"
-
-    - stage: test_windows
-      displayName: Test ACN Windows
-      dependsOn:
-        - setup
-      jobs:
-        - job: test
-          displayName: Run Tests
-          variables:
-            STORAGE_ID: $[ stagedependencies.setup.env.outputs['EnvironmentalVariables.StorageID'] ]
-          pool:
-            name: "$(BUILD_POOL_NAME_DEFAULT_WINDOWS_ALT)"
-          steps:
-            - script: |
-                cd npm/
-                go test ./...
-              retryCountOnTaskFailure: 3
-              name: "TestWindows"
-              displayName: "Run Windows Tests"
 
-  - ${{ else }}:
-    - stage: test
-      displayName: Test ACN
-      dependsOn:
-        - setup
-      jobs:
-        - job: test
-          displayName: Run Tests
-          variables:
-            STORAGE_ID: $[ stagedependencies.setup.env.outputs['EnvironmentalVariables.StorageID'] ]
-          pool:
-            name: "$(BUILD_POOL_NAME_DEFAULT)"
-          steps:
-            - script: |
-                make tools
-                # run test, echo exit status code to fd 3, pipe output from test to tee, which splits output to stdout and go-junit-report (which converts test output to report.xml), stdout from tee is redirected to fd 4. Take output written to fd 3 (which is the exit code of test), redirect to stdout, pipe to read from stdout then exit with that status code. Read all output from fd 4 (output from tee) and write to top stdout
-                { { { {
-                      sudo -E env "PATH=$PATH" make test-all;
-                      echo $? >&3;
-                      } | tee >(build/tools/bin/go-junit-report > report.xml) >&4;
-                    } 3>&1;
-                  } | { read xs; exit $xs; }
-                } 4>&1
-              retryCountOnTaskFailure: 3
-              name: "Test"
-              displayName: "Run Tests"
-
-    - stage: test_windows
-      displayName: Test ACN Windows
-      dependsOn:
-        - setup
-      jobs:
-        - job: test
-          displayName: Run Tests
-          variables:
-            STORAGE_ID: $[ stagedependencies.setup.env.outputs['EnvironmentalVariables.StorageID'] ]
-          pool:
-            name: "$(BUILD_POOL_NAME_DEFAULT_WINDOWS_ALT)"
-          steps:
-            - script: |
-                cd npm/
-                go test ./...
-              retryCountOnTaskFailure: 3
-              name: "TestWindows"
-              displayName: "Run Windows Tests"
+  - template: templates/run-unit-tests.yaml
+    parameters:
+      poolName: "$(BUILD_POOL_NAME_DEFAULT)"
```

Comment:
Any reason for passing this in?

Reply:
isn't the pool name a parameter used in the template when we specify the job? Previously the pool name parameter was passed into the "test" job and so wanted to keep similar functionality.

Reply:
It does not have to be. 

In the original code it was not passing a parameter but a pipeline variable. That same variable, as long as it is called within the same pipeline will match. The only reason we would want to add a parameter is if we want to make the UT file dynamic on where it can be used, on which pool, ect.. (which there is no need for right now).